examples/test_dqgmres.jl

```diff
@@ -0,0 +1,51 @@
+using Krylov, LinearOperators, ILU0
+using LinearAlgebra, Printf
+
+n = 64
+N = n^3
+A = spdiagm(
+      -1 => fill(-1.0, n - 1), 
+       0 => fill(3.0, n), 
+       1 => fill(-2.0, n - 1)
+    )
+Id = sparse(1.0I, n, n)
+A = kron(A, Id) + kron(Id, A)
+A = kron(A, Id) + kron(Id, A)
+x = ones(N)
+b = A * x
+F = ilu0(A)
+
+@printf("nnz(ILU) / nnz(A): %7.1e\n", nnz(F) / nnz(A))
+
+# Solve Ax = b with DQGMRES and an ILU(0) preconditioner
+# Remark: DIOM can be used in the same way
+yM = zeros(N)
+yN = zeros(N)
+yP = zeros(N)
+opM = LinearOperator(Float64, N, N, false, false, y -> forward_substitution(yM, F, y))
+opN = LinearOperator(Float64, N, N, false, false, y -> backward_substitution(yN, F, y))
```

Comment:
```suggestion
opM = LinearOperator(Float64, N, N, false, false, y -> forward_substitution!(yM, F, y))
opN = LinearOperator(Float64, N, N, false, false, y -> backward_substitution!(yN, F, y))
```

Function rename. See https://github.com/mcovalt/ILUZero.jl/commit/a32ddc54ea97328e55de5ea1e901301614ec10fb